backend/src/plugins/fastify-roles/README.md

```diff
@@ -0,0 +1,17 @@
+# Fastify Roles plugin
+
+## Description
+The role of this plugin, might change in the future, but it does the following.
+* gets the user from the request token
+* adds the user to the request object so that the controllers have a user object with capabilities.
+* the payload now becomes an object with attributes of 
+```
+{
+    data: payload,
+    permission: '{none|read|write}',
+    user: {
+        role: '{admin|manager|gdx|subscriber|none}, /* This is the database role */
+        roles: [ pmo-sys-admin ], /* This is the keycloak role */
```

Comment:
I just want to make 100%  sure we're not relying on being able to dynamically set keycloak roles from the app. As far as I know, this is not possible due to a constraint in keycloak's configuration which is beyond our control. Keycloak roles would only ever be read-only as far as the API goes. Any roles we want to assign to a user would need to be done within the keycloak interface.

So, I'm guessing that a role like "pmo-sys-admin" would be a super admin role only given to the devs, basically?

Reply:
This is not what this plugin is doing, it is the reverse, it is getting info from keycloak, and then the goal is to get information from the db, and attach it is an object, so the controllers are aware of the capabilities a user has